Modules/DSCParser/Modules/DSCParser.psm1

```diff
@@ -177,34 +254,54 @@ function ConvertTo-DSCObject
     $ParseErrors = $null
 
     # Use the AST to parse the DSC configuration
-    if (-not [System.String]::IsNullOrEmpty($Path))
-    {
-        $AST = [System.Management.Automation.Language.Parser]::ParseFile((Resolve-Path $Path), [ref]$Tokens, [ref]$ParseErrors)
-    }
-    else
+    if (-not [System.String]::IsNullOrEmpty($Path) -and [System.String]::IsNullOrEmpty($Content))
     {
-        $AST = [System.Management.Automation.Language.Parser]::ParseInput($Content, [ref]$Tokens, [ref]$ParseErrors)
+        $Content = Get-Content $Path -Raw
     }
+    $AST = [System.Management.Automation.Language.Parser]::ParseInput($Content, [ref]$Tokens, [ref]$ParseErrors)
+    
     # Look up the Configuration definition ("")
     $Config = $AST.Find({$Args[0].GetType().Name -eq 'ConfigurationDefinitionAst'}, $False)
 
     # Retrieve information about the DSC Modules imported in the config
     # and get the list of their associated resources.
-    $DSCResources = @()
+    $ModulesToLoad = @()
     foreach ($statement in $config.body.ScriptBlock.EndBlock.Statements)
     {
         if ($null -ne $statement.CommandElements -and $null -ne $statement.CommandElements[0].Value -and `
             $statement.CommandElements[0].Value -eq 'Import-DSCResource')
         {
-            for ($i = 0; $i -le $statement.CommandElements.COunt; $i++)
+            $currentModule = @{}
+            for ($i = 0; $i -le $statement.CommandElements.Count; $i++)
             {
                 if ($statement.CommandElements[$i].ParameterName -eq 'ModuleName' -and `
                     ($i+1) -lt $statement.CommandElements.Count)
                 {         
                     $moduleName = $statement.CommandElements[$i+1].Value      
-                    $DSCResources += Get-DSCResource -Module $moduleName
+                    $currentModule.Add('ModuleName', $moduleName)
+                }
+                elseif ($statement.CommandElements[$i].ParameterName -eq 'ModuleVersion' -and `
+                    ($i+1) -lt $statement.CommandElements.Count)
+                {
+                    $moduleVersion = $statement.CommandElements[$i+1].Value 
+                    $currentModule.Add('ModuleVersion', $moduleVersion)
                 }
             }
+            $ModulesToLoad += $currentModule
+        }
+    }
+    $DSCResources = @()
+    foreach ($moduleToLoad in $ModulesToLoad)
+    {
+        $loadedModuleTest = Get-Module -Name $moduleToLoad.ModuleName -ListAvailable | Where-Object -FilterScript {$_.Version -eq $moduleToLoad.ModuleVersion}
```

Comment:
This will fail if `Import-DscResource` does not have any `ModuleVersion` set. Would be better to either load the specified or the latest module version.

Reply:
Good catch. Fixed